conans/test/integration/toolchains/meson/test_mesontoolchain.py

```diff
@@ -281,3 +281,38 @@ def test_check_c_cpp_ld_list_formats():
     assert "c = ['aarch64-poky-linux-gcc', '-mcpu=cortex-a53', '-march=armv8-a+crc+crypto']" in content
     assert "cpp = ['aarch64-poky-linux-g++', '-mcpu=cortex-a53', '-march=armv8-a+crc+crypto']" in content
     assert "ld = ['aarch64-poky-linux-ld', '--sysroot=/opt/sysroots/cortexa53-crypto-poky-linux']" in content
+
+
+@pytest.mark.skipif(platform.system() != "Darwin", reason="Requires OSX")
+def test_apple_meson_add_min_version_flag():
```

Comment:
I think there is a problem, and there are insufficient tests of ``MesonToolchain`` with different ``os.sdk`` and ``os.sdk_version`` values, for cross-building. I cannot find any coverage of these cases in the test suite, and I am concerned, following the code paths, that something might not be ok.

Reply:
I don't think so. You can have a look at the `conans/test/functional/toolchains/meson/test_cross_compilation.py` module.

Reply:
The test that is covering it has:
```
t.run("install . --profile:build=default --profile:host=profile_host")
t.run("build .")
```
That is legacy 1.X syntax and behavior, that means that the test is not really cross-compiling, and just using the ``--profile:host=default`` to build the application! And this is the test that covers the cases of ``os_sdk`` and ``os_version``

Reply:
Maybe fixing this in the test make it fail without this PR fix?

Reply:
You're completely right, but the test was cross-building because of the info gathered by the MesonToolchain` in the `conan_meson_cross.ini`. Despite that, I'm changing this test to verify if it was failing as you said.